src/shell/Shell.cpp

```diff
@@ -645,6 +644,17 @@ int main(int argc, char* argv[])
                     context->initDebugger(nullptr);
                     continue;
                 }
+                if (strcmp(argv[i], "--debugger-wait-source") == 0) {
+                    StringRef* sourceName;
+                    StringRef* clientSourceRef = context->getClientSource(&sourceName);
```

Comment:
Currently it is not possible to know whether an empty string or nothing is received, because in both cases we get an empty string. We could think about returning with a nullptr if no sources is received (ESCARGOT_DEBUGGER_THERE_WAS_NO_SOURCE is received).